### PR TITLE
[Privatization]  Skip variable assign append on ChangeReadOnlyVariableWithDefaultValueToConstantRector

### LINKED ISSUE #7900: Incorrect behavior of ChangeReadOnlyVariableWithDefaultValueToConstantRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/a8c555df-13ab-46b2-abfd-d725cbe72a94

```php
<?php

final class DemoFile
{
    /**
     * @var string[]
     */
    private const VALUES = ['a', 'b', 'c', 'd', 'e', 'a', 'b', 'c', 'd', 'e'];
    
    public function run()
    {
        $arr = [];
        foreach (self::VALUES as $key => $value) {
            $arr[$value] ??= 0;
            $arr[$value] += $key;
        }
        return $arr;
    }
}
```

### Responsible rules

* `ChangeReadOnlyVariableWithDefaultValueToConstantRector`

## Expected Behavior

Rector shouldn't suggest to convert a local variable into a private const when the local variable is just a temporary one. When the run method is executed twice, the result is not the same.

### PR DESCRIPTION
**Pull request by mickeytodd**
Closes https://github.com/rectorphp/rector/issues/7900

### CODE REVIEW

**Review by samsonasik — CHANGES_REQUESTED** [review]
Please add fixture test under

https://github.com/rectorphp/rector-src/tree/main/rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture

If it going to skip the change, you can copy one of fixture file with `skip_` prefix and modify

File: rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_variable_assignment.php.inc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Rector\Tests\Privatization\Rector\Class_\ChangeReadOnlyVariableWithDefaultValueToConstantRector\Fixture;
+
+final class SkipVariableAssignment
```
Comment (samsonasik) [inline rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_variable_assignment.php.inc]:
```suggestion
final class SkipVariableArrayAppend
```

File: rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_variable_assignment.php.inc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/Privatization/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector/Fixture/skip_variable_assignment.php.inc]: Please rename file to `skip_variable_array_append.php.inc`

**Review by samsonasik — CHANGES_REQUESTED** [review]
Please rename PR title to something like:

```
[Privatization]  Skip variable assign append on ChangeReadOnlyVariableWithDefaultValueToConstantRector
```

Also, could you make CI green, see error 

https://github.com/rectorphp/rector-src/actions/runs/4803271285/jobs/8547601933?pr=3687#step:5:18

**Comment by mickeytodd** [comment]
@samsonasik could you please check why testing fails on Renaming/Rector/FileWithoutNamespace/PseudoNamespaceToNamespaceRector/Fixture/namespace_split.php.inc? I didn't change that file. Thanks!

**Comment by samsonasik** [comment]
I am not sure, could you try rebase? If still happen, try revert your change, if still happen, it possibly due to duplicated fixture files

**Comment by samsonasik** [comment]
Test failure fixed at https://github.com/rectorphp/rector-src/pull/3691, please rebase

**Review by samsonasik — APPROVED** [review]
Looks good, thank you 👍

**Comment by plenka** [comment]
@mickeytodd @samsonasik Thanks for fixing this so quickly!

**Pull request merged**